tex/chapters/06-MachineLearning.tex

```diff
@@ -0,0 +1,240 @@
+
+\begin{multicols}{3}
+\begin{enumerate}
+  \item Evolved HMM 5
+  \item Evolved ANN
+  \item Evolved ANN 5 Noise 05
+  \item PSO Gambler 2\_2\_2 Noise 05
+  \item Nice Meta Winner Ensemble
+  \item NMWE Long Memory
+  \item EvolvedLookerUp1\_1\_1
+  \item Shubik
+  \item Retaliate (0.05)
+  \item Soft Joss: 0.9
+  \item GTFT: 0.3
+  \item ShortMem
+  \item Adaptive Pavlov 2011
+  \item Punisher
+  \item Grumpy
+  \item Hard Tit For Tat
+  \item Appeaser
+  \item Meta Majority Finite Memory
+  \item Average Copier
+  \item Random Hunter
+  \item Meta Hunter Aggressive
+  \item Worse and Worse 2
+  \item Predator
+  \item Stochastic Cooperator
+  \item Knowledgeable Worse and Worse
+  \item Meta Winner Finite Memory
+  \item Cautious QLearner
+  \item Random: 0.1
+  \item Meta Mixer
+  \item EasyGo
+  \item Hard Go By Majority: 40
+  \item ZD-Extort-2 v2
+  \item Opposite Grudger
+  \item Alternator
+  \item Tricky Defector
+  \item Negation
+  \item Gradual Killer
+  \item Cycler CCCCCD
+\end{enumerate}
+\end{multicols}
+
+In order to properly assess the model's performance, more information is required regarding what it can accurately predict and what mistakes are made.
+To do this a Confusion Matrix is constructed, as shown in Figure \ref{fig:conf_matrix}.
+
+\begin{figure}[htbp!]
+    \centering
+    \includegraphics[width=0.5\textwidth]{../img/ML/confusion_matrix.png}
+    \caption{Confusion Matrix for SVC Model}
+    \label{fig:conf_matrix}
+\end{figure}
+
+Again, it is important to consider the assumption that has been made.
+For the True Label axis, strategies with the same name are considered equivalent and strategies with different names are assumed to be different.
+The Confusion Matrix shows that the model predicted the vast majority of labels correctly (values on the main diagonal).
+No strategies that were assumed to be the same were predicted to be different by the model.
+This is encouraging, as due to the assumption it is known that all strategies assumed equal are in fact identical.
+However, the model did predict that some strategies were equivalent, despite being assumed different.
+Again, this is to be expected as no allowance was made for identical strategies with different names.
+
+Finally, it is useful to obtain a measure of the significance of each of the variables used to train the model.
+Feature selection is where the most significant (predictive) variable are filtered from the data in order to make better predictions faster \cite{Guyon2003}.
+The $\chi^2$ test is commonly used for performing feature selection on models \cite{Forman2003} by comparing the distribution of each feature with the independent variable.
+A $\chi^2$ test was used to obtain the p Values shown in Table \ref{tab:pvalues} which shows that all of the variables used are statistically significant and should be included in model.
+
+\begin{table}[htbp!]
+    \centering
+    \includestandalone[]{../img/p-values}
```

Comment:
Better to keep this in a `tab` folder (to make it clear that it's not an image).